pkg/build_zip.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This tool builds zip files from a list of inputs."""
+
+import sys
+
+from absl import flags
+from helpers import UnquoteAndSplit
+from zipfile import ZipFile, ZipInfo, ZIP_DEFLATED
+
+flags.DEFINE_string('output', None, 'The output file, mandatory')
+flags.mark_flag_as_required('output')
+
+flags.DEFINE_multi_string('file', [], 'A file to add to the zip')
+
+FLAGS = flags.FLAGS
+
+def main(unused_argv):
+  with ZipFile(FLAGS.output, 'w') as zip:
+    for f in FLAGS.file:
+      (src_path, dst_path) = UnquoteAndSplit(f, '=')
+
+      # remove leading /'s: the zip format spec says paths should never have a
+      # leading slash, but Python will happily do this. The built-in zip tool
+      # in Windows will complain that such a zip file is invalid.
+      if dst_path[0] == '/':
+          dst_path = dst_path[1:]
+
+      entry_info = ZipInfo(
+        filename = dst_path,
+        # choose something deterministic.
+        # this is the earliest date possible in the basic zip file format.
+        date_time = (1980, 1, 1, 0, 0, 0),
+      )
+
+      entry_info.compress_type = ZIP_DEFLATED
```

Comment:
The docs say not every distribution of Python has to support this, but every one I tested did. The other options are [`ZIP_STORED`](https://docs.python.org/3/library/zipfile.html#zipfile.ZIP_STORED) (no compression) and two other compression formats which are apparently not well-supported by tools.

I didn't make an attr for compression level (ala #60). Do you want it in this PR or a future one?

Reply:
We can do this in a followup. There is a lot of other scaffolding to do first.